### DISCUSSION TITLE
Any issues using it with Kotlin and Gradle?

### DISCUSSION THREAD

**Discussion by Anksmahajan** [spring-projects/spring-modulith#238] [General]
We are planning on a new service but want to use Kotlin and Gradle. All examples I see use Java and Maven. I am wondering if folks have come across any issues I should be aware of.

**Comment by genuss** [comment]
I have an experience with using spring modulith with kotlin and gradle. I didn't use many features, only @ApplicationModuleTest. It works without an issue.
BTW, I don't see any possible issues with gradle as it's just a build tool and modulith doesn't intercept build process. There might be something with kotlin, maybe with coroutines, but I didn't use them in my project, so I can't say for sure.

**Comment by domgom** [comment]
Well, because Kotlin's default visibility is public, you'll find yourself writing a lot of "private" keywords. 
I usually organise my domains in a single .kt file inside the package so it's a bit tedius. You can always have a subpackage ("impl", "internal" or whatever) and that solves the issue.
